mpp/include/mpp_define_nest_domains.inc

```diff
@@ -60,112 +62,330 @@ subroutine mpp_define_nest_domains(nest_domain, domain_fine, domain_coarse, tile
+           exit
+        end if
+     end do
+     if(my_num_nest .NE. 1) then
+        print*, "num_nest=", my_num_nest, nest_domain%domain_fine%tile_id(1), nest_domain%tile_fine(1)
+        call mpp_error(FATAL, "mpp_define_nest_domains.inc: my_num_nest .ne. 1 on fine pelist")
      endif
-  endif
-  ! First computing the send and recv information from find to coarse.
-  if( npes == npes_coarse ) then
-     concurrent = .false.
-  else if( npes_fine + npes_coarse == npes ) then
-     concurrent = .true.
-  else
-     call mpp_error(FATAL, "mpp_domains_define.inc: size(pelist_coarse) .NE. size(pelist) and "// &
-              "size(pelist_coarse)+size(pelist_fine) .NE. size(pelist)")
+  else if( nest_domain%is_coarse_pe ) then
+     my_num_nest = nest_domain%num_nest
+     do n = 1, nest_domain%num_nest
+        my_nest_id(n) = n
+     enddo
   endif
 
-  !--- to confirm integer refinement.
-  nx_coarse = iend_coarse - istart_coarse + 1
-  ny_coarse = jend_coarse - jstart_coarse + 1
-  nx_fine   = iend_fine - istart_fine + 1
-  ny_fine   = jend_fine - jstart_fine + 1
+  nest_domain%my_num_nest = my_num_nest
+  if(my_num_nest>0) then
+     allocate(nest_domain%my_nest_id(my_num_nest))
+     nest_domain%my_nest_id(:) = my_nest_id(1:my_num_nest)
+  endif
 
-  if( mod(nx_fine,nx_coarse) .NE. 0 ) call mpp_error(FATAL, &
-      "mpp_domains_define.inc: The refinement in x-direction is not integer for nest domain"//trim(nest_domain%name) )
-  x_refine = nx_fine/nx_coarse
-  if( mod(ny_fine,ny_coarse) .NE. 0 ) call mpp_error(FATAL, &
-      "mpp_domains_define.inc: The refinement in y-direction is not integer for nest domain"//trim(nest_domain%name) )
-  y_refine = ny_fine/ny_coarse
+  !--- We are assuming the fine grid is fully overlapped with coarse grid.
+  if( nest_domain%is_fine_pe ) then
+     if( nest_domain%iend_fine(my_nest_id(1))-nest_domain%istart_fine(my_nest_id(1))+1 &
+         .NE. nest_domain%domain_fine%x(1)%global%size .OR.  &
+         nest_domain%jend_fine(my_nest_id(1))-nest_domain%jstart_fine(my_nest_id(1))+1 &
+         .NE. nest_domain%domain_fine%y(1)%global%size ) then
+        print*, "x size are", nest_domain%domain_fine%x(1)%global%size, &
+                   nest_domain%istart_fine(my_nest_id(1)), nest_domain%iend_fine(my_nest_id(1)) 
+        print*, "y size are", nest_domain%domain_fine%y(1)%global%size, &
+                   nest_domain%jstart_fine(my_nest_id(1)), nest_domain%jend_fine(my_nest_id(1))
+        call mpp_error(FATAL, "mpp_define_nest_domains.inc: The fine global domain is not covered by coarse domain")
+     endif
+  endif
 
-  !--- coarse grid and fine grid should be both symmetry or non-symmetry.
-  if(domain_coarse%symmetry .AND. .NOT. domain_fine%symmetry) then
-     call mpp_error(FATAL, "mpp_domains_define.inc: coarse grid domain is symmetric, fine grid domain is not")
+  ! only support concurrent run for fine and coarse domain, currently only check on coarse pe
```

Comment:
Is this entire block supposed to be commented out?

Reply:
I will comment out the extra two lines, but I want to leave the test code in place for now.